### PR TITLE
add opa config file to load bundles 

### PR DESCRIPTION
**Pull request by superolo92**
- start opa with config to load bundles
 - update namespace policy
 - update namespace test fixtures

### CODE REVIEW

File: manifests/opa.yaml [hunk 1]
```diff
@@ -152,7 +158,30 @@ spec:
...
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: opa-config
+  namespace: opa
+data:
+  config.yaml: |
+    services:
+      - name: discovery
+        url: https://discovery-opa-bundles.s3-eu-west-1.amazonaws.com
```
Comment (moshloop) [inline manifests/opa.yaml]: This needs to be externalized into a variable e.g:

```yaml
opa:
   bundles:
      discovery:
           url: ""
           prefix: ""
           name: ""
```

File: pkg/types/types.go [hunk 2]
```diff
@@ -92,9 +92,12 @@ type Calico struct {
 }
 
 type OPA struct {
-	Disabled        bool   `yaml:"disabled,omitempty"`
-	KubeMgmtVersion string `yaml:"kubeMgmtVersion,omitempty"`
-	Version         string `yaml:"version,omitempty"`
+	Disabled          bool   `yaml:"disabled,omitempty"`
+	KubeMgmtVersion   string `yaml:"kubeMgmtVersion,omitempty"`
+	Version           string `yaml:"version,omitempty"`
+	BundleUrl         string `yaml:"bundle_url,omitempty"`
```
Comment (moshloop) [inline pkg/types/types.go]: Use bundleUrl, bundlePrefix to be consistent with other field names

File: manifests/opa.yaml (hunk 1 above)
Comment (superolo92) [reply manifests/opa.yaml]: remediated,  thanks Mosh!

File: test/common.yml [hunk 3]
```diff
@@ -23,6 +23,9 @@ serviceSubnet: 100.100.0.0/16
 opa:
   version: 0.13.5
   kubeMgmtVersion: 0.8
+  bundle_url: https://discovery-opa-bundles.s3-eu-west-1.amazonaws.com
```
Comment (moshloop) [inline test/common.yml]: we can't test in this repo with discovery assets, we can host a prepacked bundle in Github and then reference it via: https://raw.githubusercontent.com/

File: manifests/opa.yaml [hunk 4]
```diff
@@ -107,10 +107,13 @@ spec:
           args:
             - "run"
             - "--server"
+            - "--config-file=/config/config.yaml"
             - "--tls-cert-file=/certs/tls.crt"
             - "--tls-private-key-file=/certs/tls.key"
             - "--addr=0.0.0.0:443"
             - "--addr=http://127.0.0.1:8181"
+            - "--log-format=json-pretty"
+            - "--set=decision_logs.console=true"
```
Comment (moshloop) [inline manifests/opa.yaml]: I think this will need to be plumbed through as well - i.e. `opa.decisionLogs: true`

File: manifests/opa.yaml [hunk 5]
```diff
@@ -152,7 +158,30 @@ spec:
...
+---
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: opa-config
+  namespace: opa
+data:
+  config.yaml: |
+    services:
+      - name: {{.opa.bundle_service_name}}
```
Comment (moshloop) [inline manifests/opa.yaml]: This is going to create an invalid config when the bundles are not set, we will need to add a check if the bundles are provided and adjust the config accordingly + add another test fixture without the bundle to verify it

File: test/opa/opa-fixtures/accepted/namespace-valid.yaml [hunk 6]
```diff
@@ -1,10 +1,8 @@
...
-  labels: 
-    team-name: bendersnatch
-    technical-lead: superolo
-  annotations:
-    ingress-whitelist: "*.acmecorp.com"
-    registry-whitelist: "nginx,grafana,prom,k8s.gcr.io"
\ No newline at end of file
+  name: superolo-world-good
+  labels:
+    company: DISCOVERY-VITALITY
```
Comment (moshloop) [inline test/opa/opa-fixtures/accepted/namespace-valid.yaml]: Use mock names, not real names - The fork's git history will also need to be rewritten

File: test/opa/policies/check-ns.rego [hunk 7]
```diff
@@ -1,21 +1,36 @@
...
+has_key(obj, k) { _ = obj[k]}
+has_item(list, i) {list[_] = i}
 
 deny[msg] {
     input.request.kind.kind = "Namespace"
     input.request.operation = "CREATE"
     metadata = input.request.object.metadata
     not has_key(metadata,"labels")
-    msg = "no lables specified"
+    msg = "No company lables specified!"
```
Comment (moshloop) [inline test/opa/policies/check-ns.rego]:
```suggestion
    msg = "Must specify label: company"
```

File: test/opa/policies/check-ns.rego [hunk 8]
```diff
@@ -1,21 +1,36 @@
...
 deny[msg] {
     input.request.kind.kind = "Namespace"
     input.request.operation = "CREATE"
+    incoming_service = input.request.object.metadata.labels.service
+    incoming_company = input.request.object.metadata.labels.company
     labels = input.request.object.metadata.labels
-    not has_key(labels,"team-name")
-    msg = sprintf("please add team-name label. your labels are: %q",[labels])
+    not has_key(ca.companies,incoming_company )
+    msg = sprintf("Your company name does not exist company: %q",[incoming_company])
```
Comment (moshloop) [inline test/opa/policies/check-ns.rego]:
```suggestion
    msg = sprintf("Company name '%q' does not exist",[incoming_company])
```

File: test/opa/policies/check-ns.rego [hunk 9]
```diff
@@ -1,21 +1,36 @@
...
+
+deny[msg] {
+    input.request.kind.kind = "Namespace"
+    input.request.operation = "CREATE"
+    incoming_service = input.request.object.metadata.labels.service
+    incoming_company = input.request.object.metadata.labels.company
+    services = ca.companies[incoming_company]
+    labels = input.request.object.metadata.labels
+    not has_item(services, incoming_service)
+    msg = sprintf("Your company cannot use this service: %q",[incoming_service])
```
Comment (moshloop) [inline test/opa/policies/check-ns.rego]:
```suggestion
    msg = sprintf("Service '%q' not found for company",[incoming_service])
```

**Pull request merged**